tests/e2e/user-journeys.spec.ts

```diff
@@ -119,4 +119,146 @@ test.describe('User journeys', () => {
       expect(response?.headers()['x-nf-edge-functions']).toBe('server')
     })
   })
+
+  test('response has user-defined Cache-Control header when using origin SSR', async ({ page, serverlessSite }) => {
+    const response = await page.goto(`${serverlessSite.url}/headers`)
+    await expect(page.getByRole('heading', { name: /Headers/i })).toBeVisible()
+    expect(response?.headers()['cache-control']).toBe('public,max-age=3600')
+  })
+
+  test('response has user-defined Cache-Control header when using edge SSR', async ({ page, edgeSite }) => {
+    const response = await page.goto(`${edgeSite.url}/headers`)
+    await expect(page.getByRole('heading', { name: /Headers/i })).toBeVisible()
+    expect(response?.headers()['cache-control']).toBe('public,max-age=3600')
+  })
+
+  test('user can configure Stale-while-revalidate when using origin SSR', async ({ page, serverlessSite }) => {
+    // attempt to purge cdn cache in case this is a retry
+    await fetch(`${serverlessSite.url}/purge-cdn?tag=stale-while-revalidate-tag`)
+
+    await new Promise((resolve) => setTimeout(resolve, 1000))
+
+    await page.goto(`${serverlessSite.url}/stale-while-revalidate`)
+    const responseGeneratedAtText1 = await page.getByText('Response generated at').textContent()
+
+    await new Promise((resolve) => setTimeout(resolve, 5000))
```

Comment:
We can replace all of these with a built-in:
```suggestion
    await page.waitForTimeout(5000);
```